backend/src/main/java/gov/cdc/usds/simplereport/config/ApiUserAwareGraphQlContextBuilder.java

```diff
@@ -0,0 +1,68 @@
+package gov.cdc.usds.simplereport.config;
+
+import gov.cdc.usds.simplereport.config.authorization.SiteAdminPrincipal;
+import gov.cdc.usds.simplereport.config.authorization.UserPermissionPrincipal;
+import gov.cdc.usds.simplereport.service.ApiUserService;
+import graphql.kickstart.execution.context.DefaultGraphQLContext;
+import graphql.kickstart.execution.context.GraphQLContext;
+import graphql.kickstart.servlet.context.DefaultGraphQLServletContext;
+import graphql.kickstart.servlet.context.DefaultGraphQLWebSocketContext;
+import graphql.kickstart.servlet.context.GraphQLServletContextBuilder;
+import java.security.Principal;
+import java.util.Collections;
+import java.util.HashSet;
+import javax.security.auth.Subject;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.websocket.Session;
+import javax.websocket.server.HandshakeRequest;
+import org.dataloader.DataLoaderRegistry;
+import org.springframework.stereotype.Component;
+
+@Component
+class ApiUserAwareGraphQlContextBuilder implements GraphQLServletContextBuilder {
+  private final ApiUserService apiUserService;
+
+  ApiUserAwareGraphQlContextBuilder(ApiUserService apiUserService) {
+    this.apiUserService = apiUserService;
+  }
+
+  @Override
+  public GraphQLContext build(
+      HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) {
+    return DefaultGraphQLServletContext.createServletContext()
+        .with(httpServletRequest)
+        .with(httpServletResponse)
+        .with(subjectFromCurrentUser())
+        .build();
+  }
+
+  @Override
+  public GraphQLContext build(Session session, HandshakeRequest handshakeRequest) {
+    return DefaultGraphQLWebSocketContext.createWebSocketContext()
+        .with(session)
+        .with(handshakeRequest)
+        .with(subjectFromCurrentUser())
+        .build();
+  }
+
+  @Override
+  public GraphQLContext build() {
+    return new DefaultGraphQLContext(new DataLoaderRegistry(), subjectFromCurrentUser());
+  }
+
+  private Subject subjectFromCurrentUser() {
+    var currentUser = apiUserService.getCurrentUserInfo();
+    var principals = new HashSet<Principal>();
+
+    if (currentUser.getIsAdmin()) {
+      principals.add(new SiteAdminPrincipal());
+    }
+
+    currentUser.getPermissions().stream()
+        .map(UserPermissionPrincipal::new)
+        .forEach(principals::add);
+
+    return new Subject(true, principals, Collections.emptySet(), Collections.emptySet());
```

Comment:
So we would have to do a little more folding and mutilating of our data to get the UserInfo to pass through intact to the AuditService call at the far end where we want to know the identity and permissions of the current user. Might still be worth it? Depends whether or not we retain the service-layer permissions checks, I guess.

Reply:
I can pass through the user's organization, roles, facilities, and `ApiUser` object along as separate principals.

Reply:
Let's wait and do that after we've gotten everything else that's actually urgent done. 😁